presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -589,6 +589,8 @@ public void prepareFinal()
 
         private ObjectBigArray<GroupIdPage> rawInputs = new ObjectBigArray<>();
         private ObjectBigArray<RowBlockBuilder> blockBuilders;
+        private long rawInputsSizeInBytes;
+        private long blockBuildersSizeInBytes;
```

Comment:
@highker probably isn't necessary to count blockBuilders in as much detail as I have done so in this PR (since it created during `evaluateIntermediate` at which point we are already spilling). I have included some logic to do this but we can strip it out if you see fit. I figured we should report this memory correctly just to avoid the scenario in which another operator overallocates.

Reply:
Also I'm all for removing this since  `rowBlockBuilder.getRetainedSizeInBytes()` is a pretty slow call (it needs to loop over all the fields and call getRetainedSize() on all of them) and it is called in a loop during blockBuilders creation.